File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        int[] arr = new int[26];
        for(int i=0; i<26; i++){
            arr[i] = -1;
        }
        for(int i=0; i<s.length(); i++){
            int index = s.charAt(i) - 'a';
            if(arr[index] == -1){
                answer[i] = -1;
            } else {
                answer[i] = i - arr[index];
            } 
            arr[index] = i;
        }
        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    arr = [-1] * 26
    for i in range(len(s)):
        index = ord(s[i]) - ord('a')
        if arr[index] == -1:
            answer.append(-1)
        else:
            answer.append(i-arr[index])
        arr[index] = i
    return answer